wms/WmsApi.php

```diff
@@ -82,233 +49,248 @@ public function readRequest(): void
-                    if ($json['is_agency_secure'] == "t"){
+                    if ($json['is_agency_secure'] == "t") {
                         $agencycnt += 1;
                     }
                 }
             }
         }
 
         if ($agencycnt == 0) return true;
-        if ($agencycnt != count($requestDataArray)){
+        if ($agencycnt != count($requestDataArray)) {
             //some layers require agency filter others don't
             //we don't support this
-            $this->apiResponse->setHttpCode(401);
-            $this->apiResponse->setFormat("application/json");
-            $this->apiResponse->setBody('{"error": "Cannot combine layers that require agency validation with those that do not."}');
+            $this->error(401, "Cannot combine layers that require agency validation with those that do not.");
             return false;
-        } 
-        
+        }
+
         //add agency id to cql filter
-        $ids=null;
+        $ids = null;
         $hasall = false;
         foreach ($requestDataArray as $data) {
-            $agencies = $this->user->getAgencies($data, "read"); 
-            if ($hasall == true && $agencies != null){
-                $this->apiResponse->setHttpCode(401);
-                $this->apiResponse->setFormat("application/json");
-                $this->apiResponse->setBody('{"error": "Cannot combine layers that have difference agency access."}');
+            $agencies = $this->user->getAgencies($data, "read");
+            if ($hasall == true && $agencies != null) {
+                $this->error(401, "Cannot combine layers that have difference agency access.");
                 return;
             }
-            if ($agencies == null){
+            if ($agencies == null) {
                 $hasall = true;
             }
-            
-            if ($ids == null){
+
+            if ($ids == null) {
                 $ids = $agencies;
-            }else if ($ids !== $agencies){
-                $this->apiResponse->setHttpCode(401);
-                $this->apiResponse->setFormat("application/json");
-                $this->apiResponse->setBody('{"error": "Cannot combine layers that have difference agency access."}');
+            } else if ($ids !== $agencies) {
+                $this->error(401, "Cannot combine layers that have difference agency access.");
                 return;
             }
         }
-        
-        if ($ids == null){
+
+        if ($ids == null) {
             //has access to all data
             return true;
-        } 
-            
-        if ($this->cqlfilter == ""){
+        }
+
+        if ($this->cqlfilter == "") {
             $this->cqlfilter = "agency_id in (" . implode(",", $ids) . ")";
-        }else{
+        } else {
             $this->cqlfilter = "(" . $this->cqlfilter . ") and agency_id in (" . implode(",", $ids) . ")";
         }
         return true;
     }
 
+    private function handleGetCapabilities($requestURL, $arrContextOptions)
```

Comment:
can any of this function be shared between wms/wfs and moved into the new OWS class?

Reply:
The internals of the capabilities handling are different enough that this doesn't really make sense to do.